src/ActivityStickyQueues/FileProcessingWorkflow.cs

```diff
@@ -0,0 +1,71 @@
+using Microsoft.Extensions.Logging;
+using Temporalio.Workflows;
+
+namespace TemporalioSamples.ActivityStickyQueues;
+
+[Workflow]
+public class FileProcessingWorkflow
+{
+    [WorkflowRun]
+    public async Task RunAsync(int maxAttempts = 5)
+    {
+        for (var attempt = 1; attempt <= maxAttempts; ++attempt)
+        {
+            try
+            {
+                var uniqueWorkerTaskQueue = await Workflow.ExecuteActivityAsync(
+                    (NonStickyActivities act) => act.GetUniqueTaskQueueAsync(),
+                    new()
+                    {
+                        StartToCloseTimeout = TimeSpan.FromMinutes(1),
+                    });
+
+                var downloadPath = Path.GetTempFileName();
```

Comment:
This is non-deterministic. Please move this to the activity (i.e. maybe the first activity needs to return this)

Reply:
Oh yes good catch

Reply:
I updated it to this: `var downloadPath = Path.Join(Path.GetTempPath(), Workflow.NewGuid().ToString());` 

`Path.GetTempPath` should be deterministic right?

Reply:
The docs say: "Returns the path of the current user's temporary folder."

Reply:
That's not deterministic if it is current-user (or even machine) specific. You should accept this as a response from the first activity and let that activity choose to tell you where they downloaded the file. This also is better practice anyways to let the downloaded choose the temporary location for the file.